config/config.yaml

```diff
@@ -0,0 +1,148 @@
+defaults:
+  region: {{ .ctx.region}}
+  # Resourcegroups
+  globalRG: global
+  regionRG: hcp-underlay-{{ .ctx.region }}-{{ .ctx.regionStamp}}
+  serviceClusterRG: hcp-underlay-{{ .ctx.region }}-svc-{{ .ctx.regionStamp }}
+  managementClusterRG: hcp-underlay-{{ .ctx.region }}-mgmt-{{ .ctx.regionStamp }}-{{ .ctx.cxStamp }}
+  imageSyncRG: hcp-underlay-{{ .ctx.region }}-imagesync-{{ .ctx.regionStamp }}
+
+  # General AKS config
+  kubernetesVersion: 1.30.5
+  istioVersion: "['asm-1-22']"
+  vnetAddressPrefix: "10.128.0.0/14"
+  persist: true
+  subnetPrefix: "10.128.8.0/21"
+  podSubnetPrefix: "10.128.64.0/18"
+  aksName: aro-hcp-aks
+
+  # ACR
+  serviceComponentAcrResourceGroups: global
+
+  # SVC cluster specifics
+  svcEtcdKVName: {{ azureKeyVaultName "aro-hcp-etcd" 5 .ctx.region .ctx.regionStamp }}
+  svcEtcdKVSoftDelete: true
+
+  # MGMT cluster specifics
+  mgmtEtcdKVName: {{ azureKeyVaultName "aro-hcp-etcd" 5 .ctx.region .ctx.regionStamp .ctx.cxStamp }}
+  mgmtEtcdKVSoftDelete: true
+
+  # Frontend
+  frontendCosmosDBDeploy: true
+  frontendCosmosDBDisableLocalAuth: true
+  frontendCosmosDBName: {{ azureCosmosDBName "aro-hcp-rp" 5 .ctx.region .ctx.regionStamp }}
+
+  # Maestro
+  maestroKeyVaultName: {{ azureKeyVaultName "maestro" 5 .ctx.region .ctx.regionStamp }}
```

Comment:
as already discussed, I think a nested dictionary would be better here.
I.e. 
```
maestro:
   KeyVaultName: {{ azureKeyVaultName "maestro" 5 .ctx.region .ctx.regionStamp }}
   EventgridName: {{ azureEventGridName "maestro" 5 .ctx.region .ctx.regionStamp }}
   EventGridMaxClientSessionsPerAuthName: '4'
....
```

Reply:
agreed. i will investigate how we need to adapt the tools and will refactor in a followup PR